Improving const correctness for core MOOSE systems

permcody opened the pull request:
closes #10877

permcody committed:
Improving const correctness for core MOOSE systems

closes #10877

YaqiWang commented:
Can `FEProblemBase:: getUserObject` be const?

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/160396/) on 7c008cd wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/10895/site)

This comment will be updated on new commits.

permcody commented:
Probably. This is already a lot of changes. There may be more available. If
you have any others you’d like me to look at, let me know.
On Fri, Feb 23, 2018 at 4:51 PM MOOSE Build <notifications@github.com>
wrote:

> Job Documentation <https://www.moosebuild.org/job/160396/> on 7c008cd
> <https://github.com/idaholab/moose/commit/7c008cd5c31d4dc61e420a1b2305b86aad4baf48>
> wanted to post the following:
>
> View the site here <http://mooseframework.org/docs/PRs/10895/site>
>
> This comment will be updated on new commits.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/10895#issuecomment-368172651>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AC5XIGffR7ocrwobs8VTguPkb3p-K9C2ks5tX077gaJpZM4SRku1>
> .
>

YaqiWang commented:
This one and the `getMaterialWarehouse` are the two prevent us from using const FEProblemBase.

permcody committed:
Getting rid of another RHS insert

While this RHS is fine and expected, I think moving towards more modern sytnax
is a better option. This code will be further improved with C++17 "structed binding declarations".
